### ISSUE TITLE
unexpected fail when using `make ut` to run the tests

### ISSUE THREAD

**Issue by tiancaiamao** [pingcap/tidb#31751]
## Enhancement

Use `make ut` to run the test cases, sometimes the code fail unexpectedly.

The child process receive unexpected `sigsegv` or `sigtrap`... and then exit
No error stack

After debugging and bisearch for while, I find it's caused by the `numactl` command.
The minimal reproduce is run this repeatly:

```
numactl --physcpubind 3 -- ./executor.test.bin -test.timeout 20s -test.cpu 1 -test.run TestWriteListPartitionTable
```

if `numactl` is not used, the child process works well:

```
./executor.test.bin -test.timeout 20s -test.cpu 1 -test.run TestWriteListPartitionTable
```

**Comment by tiancaiamao** [comment]
This should be fixed before #30822

**Comment by tiancaiamao** [comment]
Closed by https://github.com/pingcap/tidb/pull/31891
